specs/_features/maxeb_increase/capella.py

```diff
@@ -1939,6 +1982,37 @@ def process_voluntary_exit(state: BeaconState, signed_voluntary_exit: SignedVolu
     initiate_validator_exit(state, voluntary_exit.validator_index)
 
 
+def process_consolidation(state: BeaconState, signed_consolidation: SignedConsolidation) -> None:
+    consolidation = signed_consolidation.message
+    target_validator = state.validators[consolidation.target_index]
+    source_validator = state.validators[consolidation.source_index]
+    # Verify the source and the target are active
+    assert is_active_validator(source_validator)
+    assert is_active_validator(target_validator)
+    # Verify exits for source and target have not been initiated
+    assert source_validator.exit_epoch == FAR_FUTURE_EPOCH
+    assert target_validator.exit_epoch == FAR_FUTURE_EPOCH
+    # Consolidations must specify an epoch when they become valid; they are not valid before then
+    assert get_current_epoch(state) >= consolidation.epoch 
+
+    # Verify the source and the target have Execution layer withdrawal credentials
+    assert source_validator.withdrawal_credentials[:1] in (ETH1_ADDRESS_WITHDRAWAL_PREFIX, COMPOUNDING_WITHDRAWAL_PREFIX)
+    assert target_validator.withdrawal_credentials[:1] in (ETH1_ADDRESS_WITHDRAWAL_PREFIX, COMPOUNDING_WITHDRAWAL_PREFIX)
+    # Verify the same withdrawal address
```

Comment:
the same withdrawal address really does feel rather limiting. what if we just verified the source ? giving someone stake without their consent doesn't seem that likely lol

Reply:
It must, given the consolidation is signed by the active key

This could only be used for withdrawal cred change if triggered/signed by the withdrawal cred itself